lib/config-validator.ts

```diff
@@ -58,6 +58,17 @@ type PackageJson = {
   'renovate-config'?: Record<string, RenovateConfig>;
 };
 
+let isFoundConfigFile = false;
+async function validateAndSetFlag(
```

Comment:
Why not build this into the `validate()` function itself? It doesn't appear to be called anywhere except from here.

Reply:
https://github.com/renovatebot/renovate/pull/23815/files#diff-bddfe40e9d4e7c7e05c0bf264770b25b0d45489e9ff1bace3c4f06cd607ddabfR167

Reply:
Apologies for the oversight. I misunderstood your initial comment. I've now integrated the functionality of `validateAndSetFlag` into the `validate` function and set `isFoundConfigFile` directly within it. Thanks for pointing it out.